src/repositories/posts.repository.js

```diff
@@ -58,15 +58,23 @@ exports.create = async (newPost, result) => {
       }
     }
 
-    for (const item of mapAllTagsWithoutDesc) {
-      const tagDescription = await investApi.fetchTagDesc(item);
+    let mapAllTagsWithoutDescString = '';
 
-      mapNewTags.push({
-        tagname: item,
-        description: tagDescription,
-      });
+    /**
+     * prepare a string of tags with ";" as delimeter
+     * for eg:- [java, javascript] will become "java;javascript"
+     */
+    for (let i = 0; i < mapAllTagsWithoutDesc.length; i++) {
+      if (i === mapAllTagsWithoutDesc.length - 1) {
+        mapAllTagsWithoutDescString += `${mapAllTagsWithoutDesc[i]}`
+      } else {
+        mapAllTagsWithoutDescString += `${mapAllTagsWithoutDesc[i]};`
+      }
```

Comment:
Why not just use join method of javascript and avoid using loops?

Reply:
Correct, making a change.